CMakeLists.txt

```diff
@@ -86,28 +64,42 @@ set(CMAKE_SKIP_BUILD_RPATH FALSE)
 set(CMAKE_SKIP_INSTALL_RPATH FALSE)
 set(CMAKE_SKIP_RPATH FALSE)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-add_compile_options(-Wall -Wextra -Werror -Wno-unused-function -Wno-unused-parameter)
 
 if (IRODS_DISABLE_COMPILER_OPTIMIZATIONS)
-    add_compile_options(-O0)
+  add_compile_options(-O0)
+endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fpermissive>)
```

Comment:
I'm guessing this is because of `with_durability`?

Reply:
This is due to a number of things. I'm going to be opening a separate issue about it.

Reply:
Got it.